src/utils/viewer/viewer.cpp

```diff
@@ -604,6 +719,9 @@ void Viewer::keyPressEvent(KeyEvent& event) {
     case KeyEvent::Key::Esc:
       std::exit(0);
       break;
+    case KeyEvent::Key::LeftShift:
+      objectSelectionOn_ = true;
+      break;
```

Comment:
Instead of explicitly maintaining this state, you can change the check in `mousePressEvent()` to 

```cpp
if (event.button() == MouseEvent::Button::Right  && (event.modifiers() & MouseEvent::Modifier::Shift))
```

That's the designated way to do this. Then you also don't need `keyReleaseEvent()` at all.